### DISCUSSION TITLE
Should Quarto prefer the knitr rendering engine if the doc contains bash scripts?

### DISCUSSION THREAD

**Discussion by Robinlovelace** [quarto-dev/quarto-cli#3590] [Q&A] [answered]
If you start a .qmd with a `{bash}` chunk, it seems to default to using the Jupyter rendering engine. That is unhelpful because there is not output from that engine. To force knitr, I make the first chunk a hidden R command (could not recall how to enforce rendering engine). Suggestion: if knitr is available and the .qmd chunk contains bash scripts, knitr should be preferred.

Asking as a question because not 100% sure this is a bug.

**Comment by eitsupi** [comment, marked as answer]
> could not recall how to enforce rendering engine

See https://quarto.org/docs/computations/execution-options.html#engine-binding

I would always recommend including the `engine` to make it do what it is intended to do.
(If we have a Linter for qmd files, it would make sense to force the engine in the default settings.)

> Suggestion: if knitr is available and the .qmd chunk contains bash scripts, knitr should be preferred.

I have not tried it, but I don't think it is generally true because of the existence of the bash kernel.
https://github.com/takluyver/bash_kernel

**Reply by Robinlovelace** [reply]
Aha so I'd need to install the `bash_kernel` Python package to get bash outputs? Thanks for the detailed reply and links in any case, v. useful.
